I fixed some part of the failing test, but there is still one thing that I don't fully follow (that weird edge case):

```
[info] Disabling java serialization
[WARN] [SECURITY][04/05/2017 15:44:41.813] [pool-112-thread-1-ScalaTest-running-SerializationSetupSpec] [DisabledJavaSerializer(akka://addedJavaSerializationSystem)] Outgoing message attempted to use Java Serialization even though `akka.actor.allow-java-serialization = off` was set! Message type was: [class java.util.BitSet]
[info] - should throw if passed system to JavaSerializer has allow-java-serialization = off (3 milliseconds)
[info] - should have replaced java serializer *** FAILED *** (16 milliseconds)
[info]   java.lang.AssertionError: assertion failed: received unexpected message {}
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectNoMsg_internal(TestKit.scala:631)
[info]   at akka.testkit.TestKitBase$class.expectNoMsg(TestKit.scala:627)
[info]   at akka.testkit.TestKit.expectNoMsg(TestKit.scala:829)
[info]   at akka.serialization.SerializationSetupSpec$$anonfun$2$$anonfun$apply$mcV$sp$4.apply(SerializationSetupSpec.scala:142)
```

--- a/akka-actor-tests/src/test/scala/akka/serialization/SerializationSetupSpec.scala
+++ b/akka-actor-tests/src/test/scala/akka/serialization/SerializationSetupSpec.scala
@@ -5,6 +5,8 @@ package akka.serialization
 
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.atomic.AtomicInteger
+import java.util.{ BitSet ⇒ ProgrammaticJavaDummy }
+import java.util.{ Date ⇒ SerializableDummy }
 
 import akka.actor.setup.ActorSystemSetup
 import akka.actor.{ ActorSystem, BootstrapSetup, ExtendedActorSystem, Terminated }
@@ -15,8 +17,6 @@ import scala.concurrent.duration._
 
 class ConfigurationDummy
 class ProgrammaticDummy
-case class ProgrammaticJavaDummy()
-case class SerializableDummy() // since case classes are serializable
 
 /**
  * Keeps a registry of each object "serialized", returns an identifier, for every "deserialization" the original object
@@ -57,6 +57,9 @@ object SerializationSetupSpec {
       actor {
         serialize-messages = off
 
+        # this is by default on, but tests are running with off, use defaults here
+        warn-about-java-serializer-usage = on
+
         serialization-bindings {
           "akka.serialization.ConfigurationDummy" = test
         }
@@ -70,6 +73,8 @@ object SerializationSetupSpec {
     akka {
       actor {
         allow-java-serialization = off
+        # this is by default on, but tests are running with off, use defaults here
+        warn-about-java-serializer-usage = on
       }
     }
     """.stripMargin))
@@ -107,9 +112,11 @@ class SerializationSetupSpec extends AkkaSpec(
   }
   val addedJavaSerializationProgramaticallyButDisabledSettings = BootstrapSetup(None, Some(ConfigFactory.parseString("""
     akka {
-    loglevel = debug
+      loglevel = debug
       actor {
         allow-java-serialization = off
+        # this is by default on, but tests are running with off, use defaults here
+        warn-about-java-serializer-usage = on
       }
     }
     """)), None)
@@ -147,7 +154,7 @@ class SerializationSetupSpec extends AkkaSpec(
 
     "disable java serialization also for incoming messages if serializer id usually would have found the serializer" in {
       val ser1 = SerializationExtension(system)
-      val msg = SerializableDummy()
+      val msg = new SerializableDummy
       val bytes = ser1.serialize(msg).get
       val serId = ser1.findSerializerFor(msg).identifier
       ser1.findSerializerFor(msg).includeManifest should ===(false)

